### DISCUSSION TITLE
SetValue method invocation frequency

### DISCUSSION THREAD

**Discussion by georgeinva2004** [zwave-js/ZWaveJS.NET#50] [General]
Is there a limit on how often one can call the SetValue method?  Here's my problem:

My application implements "Scene" functionality by looping through a list of devices and invoking the SetValue method to set the state of the device.  I do use the await operator on the SetValue method, so I assume each method invocation completes before I call the next.  Occasionally, one of the devices in the scene does not change to the desired state.  This all started happening when I moved to the 4.0.0 branch, so I'm wondering if I have a timing problem somewhere.  Any ideas?

**Comment by marcus-j-davies** [comment]
Hi @georgeinva2004 

There shouldn't be any timing issue, if anything it should  improved 😅 .
My suggestion is to check the `success` property 

https://github.com/zwave-js/ZWaveJS.NET/blob/a45a8347f280558de2a280c982f1d1163a62363b/Visual%20Studio%20Projects/ZWaveJS.NET/ZWaveJS.NET/CMDResult.cs#L35


if it is not successfull then `ErrorCode` and `Message` should have a value.

**Reply by georgeinva2004** [reply]
Good call!!!!  Thanks
